app/main.py

```diff
@@ -1,3 +1,13 @@
 def cache(func):
-    # Write your code here
-    pass
+    cached = {}
+
+    def wrapper(*args):
+        if args not in cached:
+            cached[args] = func(*args)
+            print("Calculating new result")
+            return cached[args]
+        else:
+            print("Getting from cache")
+            return cached[args]
```

Comment:
You have a duplicate `return`, so take it outside the if-else statement; try to rewrite your decorator not to have `return` duplicated

Reply:
Done!